### PR TITLE
The move buttons improvements.

### PR DESCRIPTION
**Pull request by VladiStep**
## Description
1. Added a border for the buttons.
2. Fixed grammar mistakes.
3. Renamed new context menu options so they contain the object type name.
4. Made both buttons disable on unsupported object type selection.
5. Made the buttons show an error if the object cannot be moved.

### Notes
The first actual commit is 8ad69ea8b5a4167ab80f241c7018e80676e3000b, the previous commits are from "master".

### CODE REVIEW

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 1]
```diff
@@ -222,6 +223,8 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
                 currStoryboard.Remove(this);
             }
 
+            bool isMovable = false;
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs:226]:
```suggestion
            bool isMovable = movableTypes.Contains(obj.GetType());
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 2]
```diff
@@ -232,6 +235,22 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
...
+                        // Check if the selected game object is in the "Game objects (from all layers)" list
+                        var objectItem = GameObjItems.ItemContainerGenerator.ContainerFromItem(obj) as TreeViewItem;
+                        if (objectItem?.IsSelected != true)
+                            isMovable = true;
+                    }
+                    else
+                        isMovable = true;
+                }
+                else
+                    isMovable = movableTypes.Contains(obj.GetType());
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs:253]:
```suggestion
                if (obj is GameObject)
                {
                    var room = DataContext as UndertaleRoom;
                    if (room?.Flags.HasFlag(RoomEntryFlags.IsGMS2) == true)
                    {
                        // Check if the selected game object is in the "Game objects (from all layers)" list
                        var objectItem = GameObjItems.ItemContainerGenerator.ContainerFromItem(obj) as TreeViewItem;
                        if (objectItem?.IsSelected == true)
                            isMovable = false;
                    }
                }
                
                MoveButtonsPanel.IsEnabled = isMovable;
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 3]
```diff
@@ -277,6 +296,8 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
                         layerItem.IsSelected = true;
                 }
             }
+            
+            MoveButtonsPanel.IsEnabled = isMovable;
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 4]
```diff
@@ -1449,7 +1492,7 @@ private void MoveItem(UndertaleObject obj, int dist, bool focus = true)
             };
             if (list is null)
             {
-                Debug.WriteLine($"Can't change object position - list of selected object not found.");
+                mainWindow.ShowError("Can't change the object position - a list for the selected object is not found.");
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
                mainWindow.ShowError("Can't change the object's position - no list for the selected object was found.");
```

**Review by zivmaor — DISMISSED** [review]
minor code + grammer changes

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs (hunk 4 above)
Comment (VladiStep) [reply UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]: The "'s" shouldn't be used with inanimate objects, so it should be "Can't change a position of the object"

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs (hunk 2 above)
Comment (VladiStep) [reply UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs:253]: `room?.Flags.HasFlag(RoomEntryFlags.IsGMS2)` returns a `bool?` type, so it should be `room?.Flags.HasFlag(RoomEntryFlags.IsGMS2) == true` as it was.

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs (hunk 1 above)
Comment (VladiStep) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs:226]: `obj` doesn't exist there.

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs (hunk 2 above)
Comment (zivmaor) [reply UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs:253]: Ok. Updated

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs (hunk 4 above)
Comment (zivmaor) [reply UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]: Why? I'm not aware of such a rule in english

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 5]
```diff
@@ -232,11 +235,30 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
             {
                 ObjectEditor.Content = obj;
 
-                try
+                if (obj is View)
                 {
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 6]
```diff
@@ -232,11 +235,30 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
...
-                    if (obj is View)
-                        return;
+                    MoveButtonsPanel.IsEnabled = false;
+                    return;
+                }
+
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 7]
```diff
@@ -232,11 +235,30 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
...
+                        // Check if the selected game object is in the "Game objects (from all layers)" list
+                        var objectItem = GameObjItems.ItemContainerGenerator.ContainerFromItem(obj) as TreeViewItem;
+                        if (objectItem?.IsSelected != true)
+                            isMovable = true;
+                    }
+                    else
+                        isMovable = true;
+                }
+                else
+                    isMovable = movableTypes.Contains(obj.GetType());
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs:254]:
```suggestion

            MoveButtonsPanel.IsEnabled = isMovable;

```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 8]
```diff
@@ -277,6 +299,8 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
                         layerItem.IsSelected = true;
                 }
             }
+            
+            MoveButtonsPanel.IsEnabled = isMovable;
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 9]
```diff
@@ -232,11 +235,30 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
...
+                    }
+                    else
+                        isMovable = true;
+                }
+                else
+                    isMovable = movableTypes.Contains(obj.GetType());
 
+                try
+                {
                     DependencyObject obj1 = null;
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
                    if (obj is View)
                        return;
                        
                    DependencyObject obj1 = null;
```

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs [hunk 10]
```diff
@@ -1449,7 +1500,7 @@ private void MoveItem(UndertaleObject obj, int dist, bool focus = true)
             };
             if (list is null)
             {
-                Debug.WriteLine($"Can't change object position - list of selected object not found.");
+                mainWindow.ShowError("Can't change the object position - a list for the selected object is not found.");
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs]:
```suggestion
                mainWindow.ShowError("Can't change the object position - no list for the selected object was found.");
```

**Review by zivmaor — CHANGES_REQUESTED** [review]
minor refactoring + grammer changes

File: UndertaleModTool/Editors/UndertaleRoomEditor.xaml [hunk 11]
```diff
@@ -346,15 +346,18 @@
...
-                                        ToolTip="Move item up (-)" ToolTipService.InitialShowDelay="250"
-                                  Click="TreeViewMoveUpButton_Click" PreviewMouseLeftButtonUp="TreeViewMoveButton_MouseLeftButtonUp"/>
-                <local:RepeatButtonDark FontFamily="Marlett" FontSize="14" Content="6"
-                                        ToolTip="Move item down (+)" ToolTipService.InitialShowDelay="250"
-                                  Click="TreeViewMoveDownButton_Click" PreviewMouseLeftButtonUp="TreeViewMoveButton_MouseLeftButtonUp"/>
-            </StackPanel>
-            
+            <Border Grid.Column="1" BorderBrush="#828790" BorderThickness="0,1,1,1"
+                    Height="Auto" VerticalAlignment="Top" Margin="-3,0,0,0">
+                <StackPanel Name="MoveButtonsPanel" Margin="1,0,0,0">
```
Comment (zivmaor) [inline UndertaleModTool/Editors/UndertaleRoomEditor.xaml]:
```suggestion
                <StackPanel Name="MoveButtonsPanel" Margin="1,1,1,1">
```
I think I prefer the way it looks. Your choice.

**Review by zivmaor — APPROVED** [review]

**Pull request merged**